go/heif/heif.go

```diff
@@ -0,0 +1,699 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with heif.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package heif
+
+/*
+#cgo pkg-config: libheif
+#include <stdlib.h>
+#include <libheif/heif.h>
+*/
+import "C"
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	"io"
+	"io/ioutil"
+	"runtime"
+	"unsafe"
+)
+
+func GetVersion() string {
+	return C.GoString(C.heif_get_version())
+}
+
+type Compression int
+
+const (
+	CompressionUndefined = C.heif_compression_undefined
+	CompressionHEVC      = C.heif_compression_HEVC
+	CompressionAVC       = C.heif_compression_AVC
+	CompressionJPEG      = C.heif_compression_JPEG
+)
+
+type Chroma int
+
+const (
+	ChromaUndefined       = C.heif_chroma_undefined
+	ChromaMonochrome      = C.heif_chroma_monochrome
+	Chroma420             = C.heif_chroma_420
+	Chroma422             = C.heif_chroma_422
+	Chroma444             = C.heif_chroma_444
+	ChromaInterleavedRGB  = C.heif_chroma_interleaved_RGB
+	ChromaInterleavedRGBA = C.heif_chroma_interleaved_RGBA
+
+	ChromaInterleaved24Bit = C.heif_chroma_interleaved_24bit
+	ChromaInterleaved32Bit = C.heif_chroma_interleaved_32bit
+)
+
+type Colorspace int
+
+const (
+	ColorspaceUndefined  = C.heif_colorspace_undefined
+	ColorspaceYCbCr      = C.heif_colorspace_YCbCr
+	ColorspaceRGB        = C.heif_colorspace_RGB
+	ColorspaceMonochrome = C.heif_colorspace_monochrome
+)
+
+type Channel int
+
+const (
+	ChannelY           = C.heif_channel_Y
+	ChannelCb          = C.heif_channel_Cb
+	ChannelCr          = C.heif_channel_Cr
+	ChannelR           = C.heif_channel_R
+	ChannelG           = C.heif_channel_G
+	ChannelB           = C.heif_channel_B
+	ChannelAlpha       = C.heif_channel_Alpha
+	ChannelInterleaved = C.heif_channel_interleaved
+)
+
+type ProgressStep int
+
+const (
+	ProgressStepTotal    = C.heif_progress_step_total
+	ProgressStepLoadTile = C.heif_progress_step_load_tile
+)
+
+// --- HeifError
+
+type ErrorCode int
+
+const (
+	ErrorOk = C.heif_error_Ok
+
+	// Input file does not exist.
+	ErrorInputDoesNotExist = C.heif_error_Input_does_not_exist
+
+	// Error in input file. Corrupted or invalid content.
+	errorInvalidInput = C.heif_error_Invalid_input
+
+	// Input file type is not supported.
+	ErrorUnsupportedFiletype = C.heif_error_Unsupported_filetype
+
+	// Image requires an unsupported decoder feature.
+	ErrorUnsupportedFeature = C.heif_error_Unsupported_feature
+
+	// Library API has been used in an invalid way.
+	ErrorUsageError = C.heif_error_Usage_error
+
+	// Could not allocate enough memory.
+	ErrorMemoryAllocationError = C.heif_error_Memory_allocation_error
+
+	// The decoder plugin generated an error
+	ErrorDecoderPluginError = C.heif_error_Decoder_plugin_error
+
+	// The decoder plugin generated an error
+	ErrorEncoderPluginError = C.heif_error_Encoder_plugin_error
+
+	// Error during encoding or when writing to the output
+	// ErrorEncodingError = C.heif_error_Encoding_error
```

Comment:
@fancycode any reason this error is commented out? Was just stumbling across it.